tests/configlet/util/mock_for_switch.py

```diff
@@ -0,0 +1,186 @@
+                    missing.add(i)
+            print("Missing services: {}".format(str(missing)))
+            return False
+
+        return True
+
+
+    def get_bgp_neighbor_info(self, neighbor_ip):
+        """
+        @summary: return bgp neighbor info
+
+        @param neighbor_ip: bgp neighbor IP
+        """
+        nbip = ipaddress.ip_address(neighbor_ip)
+        if nbip.version == 4:
+            out = self.shell("vtysh -c \"show ip bgp neighbor {} json\"".format(neighbor_ip))
+        else:
+            out = self.shell("vtysh -c \"show bgp ipv6 neighbor {} json\"".format(neighbor_ip))
+
+        nbinfo = json.loads(re.sub(r"\\\"", '"', re.sub(r"\\n", "", out['stdout'])))
+        log_info("bgp neighbor {} info {}".format(neighbor_ip, nbinfo))
+
+        return nbinfo[str(neighbor_ip)]
+
+
+
+def get_duthost():
+    if os.path.exists("/etc/sonic/sonic-environment"):
+        return DutHost()
+    else:
+        return None
+
+
+def config_reload(duthost, config_source="config_db", wait=60, start_bgp=False):
+    if config_source == "config_db":
+        cmd = "config reload -y"
+    elif config_source == "minigraph":
+        cmd = "config load_minigraph -y"
+    else:
+        raise Exception("Unknown config source")
+
+    log_debug("config_reload cmd: {}".format(cmd))
+    ret = duthost.shell(cmd)
+    assert ret["rc"] == 0, "failed to run err:{}".format(str(ret["stderr"]))
+    
+    if start_bgp:
+        res = duthost.shell("config bgp startup all")
+        log_debug("config_reload started BGP")
+
+    log_debug("wait for {}".format(wait))
+    time.sleep(wait)
+    log_debug("config_reload complete")
+
+
+def wait_until(timeout, interval, delay, condition, *args, **kwargs):
+    log_debug("Wait until %s is True, timeout is %s seconds, checking interval is %s, delay is %s seconds" % \
+                    (condition.__name__, timeout, interval, delay))
+
+    if delay > 0:
+        log_debug("Delay for %s seconds first" % delay)
+        time.sleep(delay)
+
+    start_time = time.time()
+    elapsed_time = 0
+    while elapsed_time < timeout:
+        log_debug("Time elapsed: %f seconds" % elapsed_time)
+        try:
+            check_result = condition(*args, **kwargs)
+        except Exception as e:
+            exc_info = sys.exc_info()
+            details = traceback.format_exception(*exc_info)
+            log_error(
+                "Exception caught while checking {}:{}, error:{}".format(
+                    condition.__name__, "".join(details), e
+                )
+            )
+            check_result = False
+
+        if check_result:
+            log_debug("%s is True, exit early with True" % condition.__name__)
+            return True
+        else:
+            log_debug("%s is False, wait %d seconds and check again" % (condition.__name__, interval))
+            time.sleep(interval)
+            elapsed_time = time.time() - start_time
+
+    if elapsed_time >= timeout:
```

Comment:
>if elapsed_time >= timeout: [](http://example.com/codeflow?start=4&length=27)

LGTM finds a true issue: this condition will always be true. Is it expected?

Reply:
Fixed